rama-http/src/service/fs/serve_dir/open_file.rs

```diff
@@ -41,14 +45,122 @@ pub(super) struct FileOpened {
 pub(super) enum FileRequestExtent {
     Full(File, Metadata),
     Head(Metadata),
+    Embedded(Vec<u8>),
+    EmbeddedHead(u64), // Content length for HEAD requests on embedded files
+}
+
+impl FileRequestExtent {
+    pub(super) fn reader(self) -> Option<Box<dyn AsyncRead + Send + Sync + Unpin>> {
+        match self {
+            Self::Head(_) | Self::EmbeddedHead(_) => None,
+            Self::Full(file, _) => Some(Box::new(file)),
+            Self::Embedded(content) => Some(Box::new(Cursor::new(content))),
+        }
+    }
+
+    pub(super) fn range_reader_with_offset(
+        self,
+        offset: u64,
+        range_size: u64,
+    ) -> Option<Box<dyn AsyncRead + Send + Sync + Unpin>> {
```

Comment:
You should be able to return Option<impl AsyncRead + Send + Sync + Unpin> by working here internally with Either. Do let me know if I'm missing something.